rust/connlib/libs/client/src/file_logger.rs

```diff
@@ -0,0 +1,52 @@
+//! Connlib File Logger
+//!
+//! This module implements a file-based logger for connlib using tracing-appender.
+//!
+//! The log files are rotated hourly.
+//!
+//! Since these will be leaving the user's device, these logs should contain *only*
+//! the necessary debugging information, and **not** any sensitive information,
+//! including but not limited to:
+//! - WiFi SSIDs
+//! - Location information
+//! - Device names
+//! - Device serials
+//! - MAC addresses
+
+use std::path::PathBuf;
+use tracing::{level_filters::LevelFilter, Subscriber};
+use tracing_subscriber::Layer;
+
+const LOG_FILE_BASE_NAME: &str = "connlib.log";
+
+/// Create a new file logger layer.
+pub fn layer<T>(
+    log_dir: PathBuf,
+) -> (
+    Box<dyn Layer<T> + Send + Sync + 'static>,
+    tracing_appender::non_blocking::WorkerGuard,
+)
+where
+    T: Subscriber + for<'a> tracing_subscriber::registry::LookupSpan<'a>,
+{
+    tracing::info!("Saving log files to: {}", log_dir.display());
+
+    let (writer, guard) = tracing_appender::non_blocking(tracing_appender::rolling::hourly(
+        log_dir,
+        LOG_FILE_BASE_NAME,
+    ));
+
+    let layer = tracing_stackdriver::layer()
+        .with_writer(writer)
+        .with_filter(if cfg!(debug_assertions) {
+            LevelFilter::DEBUG
+        } else {
+            LevelFilter::WARN
+        })
```

Comment:
I think we should have a more customizable level-filter, this function could accept something that [`impl Filter`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/layer/trait.Filter.html)

Reply:
This way we can more easily re-use the file logger.

Reply:
Hmm, good point. How do you feel about leaving this out of the file logger entirely and letting the calling code apply the filter? E.g. in the `init_logging` function in the clients.